src/Compilers/CSharp/Portable/BoundTree/BoundDagEvaluation.cs

```diff
@@ -26,9 +36,10 @@ private Symbol Symbol
                 }
             }
         }
+
         public override int GetHashCode()
         {
-            return Hash.Combine(GetOriginalInput().GetHashCode(), this.Symbol?.GetHashCode() ?? 0);
+            return Hash.Combine(GetOriginalInput().GetHashCode(), s_ignoreEverythingComparer.GetHashCode(Symbol.GetPublicSymbol()));
```

Comment:
>s_ignoreEverythingComparer.GetHashCode(Symbol.GetPublicSymbol()) [](start = 66, length = 64)

This change is not needed. All symbols are supposed to implement GetHashCode in a way that doesn't pay attention to all aspects that could ever be ignored by different flavors of TypeCompareKind. Comparer simply delegates back to the symbol.